src/test/java/org/sonatype/nexus/ci/util/NxrmUtilTest.groovy

```diff
@@ -0,0 +1,174 @@
+      listBoxModel.get(1).name == repositories.get(0).name    //Maven Releases
+      listBoxModel.get(1).value == repositories.get(0).name
+      listBoxModel.get(2).name == repositories.get(2).name    //Maven Snapshots
+      listBoxModel.get(2).value == repositories.get(2).name
+  }
+
+  def 'it populates the list of destination repositories for NXRM2'() {
+    setup:
+      GroovyMock(RepositoryManagerClientUtil.class, global: true)
+      def nxrmConfiguration = createNxrmConfig('nxrm2')
+
+      def client = Mock(RepositoryManagerV2Client.class)
+      def repositories = [
+          [
+              id              : 'maven-releases',
+              name            : 'Maven Releases',
+              format          : 'maven2',
+              repositoryType  : 'hosted',
+              repositoryPolicy: 'Release'
+          ],
+          [
+              id              : 'maven1-releases',
+              name            : 'Maven 1 Releases',
+              format          : 'maven1',
+              repositoryType  : 'hosted',
+              repositoryPolicy: 'Release'
+          ],
+          [
+              id              : 'maven-snapshots',
+              name            : 'Maven Snapshots',
+              format          : 'maven2',
+              repositoryType  : 'hosted',
+              repositoryPolicy: 'Snapshot'
+          ],
+          [
+              id              : 'maven-proxy',
+              name            : 'Maven Proxy',
+              format          : 'maven2',
+              repositoryType  : 'proxy',
+              repositoryPolicy: 'Release'
+          ],
+          [
+              id              : 'other-maven-releases',
+              name            : 'Other Maven Releases',
+              format          : 'maven2',
+              repositoryType  : 'hosted',
+              repositoryPolicy: 'Release'
+          ]
+      ]
+      client.getRepositoryList() >> repositories
+      RepositoryManagerClientUtil.nexus2Client(nxrmConfiguration.serverUrl, nxrmConfiguration.credentialsId) >> client
+
+    when: 'retrieve the list of repositories to populate the list box'
+      def listBoxModel = NxrmUtil.doFillNexusRepositoryIdItems('nxrm2')
+
+    then: 'ListBox has the correct size'
+      //3 in total because of the empty listbox name plus 2 maven hosted release repos
+      listBoxModel.size() == 3
+
+    and: 'ListBox has empty item'
+      listBoxModel.get(0).name == FormUtil.EMPTY_LIST_BOX_NAME
+      listBoxModel.get(0).value == FormUtil.EMPTY_LIST_BOX_VALUE
+
+    and: 'ListBox is populated with maven hosted repos only'
+      listBoxModel.get(1).name == repositories.get(0).name    //Maven Releases
+      listBoxModel.get(1).value == repositories.get(0).id
+      listBoxModel.get(2).name == repositories.get(4).name    //Other Maven Releases
+      listBoxModel.get(2).value == repositories.get(4).id
+  }
+
+  //get a config, defaults to nxrm3
+  def createNxrmConfig(String id = 'nxrm3') {
```

Comment:
If we return `NxrmConfiguration` here then the type can be inferred in the calling code.